### PR TITLE
feat: border styling

### PR DESCRIPTION
**Pull request by niekvanstaveren**
Draft for border styling. Will wait until color expressions is in so we can merge that

EDIT: Merged so ready for review

### CODE REVIEW

File: src/utils/style-formatter.js [hunk 1]
```diff
@@ -55,6 +55,14 @@ export default {
       );
       styles += formatProperty('background-repeat', 'no-repeat');
     }
+    if (style.border.isUsed) {
+      // TODO add color resolver for expression
+      styles += `border: ${style.border.width}px solid ${themeResolver.resolveColor(style.border.color, palette)};`;
+      const lengthShortSide = button.clientHeight < button.clientWidth ? button.clientHeight : button.clientWidth;
```
Comment (haxxmaxx) [inline src/utils/style-formatter.js]: I'd say deconstruct width and height

File: src/ext.js [hunk 2]
```diff
@@ -393,6 +393,90 @@ export default function ext({ translator }) {
...
+                      component: 'slider',
+                      translation: 'properties.border.radius',
+                      type: 'number',
+                      ref: 'style.border.radius',
+                      min: 0,
+                      max: 100,
+                      step: 1,
+                    },
+                    borderWidth: {
+                      component: 'slider',
```
Comment (haxxmaxx) [inline src/ext.js]: Not doing this in pixels with an input field?

File: src/ext.js [hunk 3]
```diff
@@ -393,6 +393,90 @@ export default function ext({ translator }) {
...
+                      translation: 'properties.border.width',
+                      min: 0,
+                      max: 100,
+                    },
+                    colorDropdown: {
+                      type: 'string',
+                      component: 'dropdown',
+                      translation: 'properties.border.color',
+                      ref: 'style.border.useExpression',
+                      options: [
```
Comment (haxxmaxx) [inline src/ext.js]: Use `colorOptions`

File: src/ext.js (hunk 2 above)
Comment (niekvanstaveren) [reply src/ext.js]: No we want to have it responsive. I will look to change the font size as well to have it responsive, but that might need some more work

File: src/utils/style-formatter.js [hunk 4]
```diff
@@ -65,6 +63,19 @@ export default {
       );
       styles += formatProperty('background-repeat', 'no-repeat');
     }
+    if (style.border.isUsed) {
+      const { width, useExpression, radius, color, colorByExpression } = style.border;
```
Comment (haxxmaxx) [inline src/utils/style-formatter.js]: `colorByExpression` is a bit misleading, I suggest colorExpression

File: src/utils/style-formatter.js [hunk 5]
```diff
@@ -25,29 +25,27 @@ const backgroundPosition = {
 
 const formatProperty = (path, setting) => `${path}: ${setting};`;
 
-const formatColorProperty = (path, inputColor, defaultColor, isExpression) => {
+const formatColorProperty = (inputColor, defaultColor, isExpression) => {
```
Comment (haxxmaxx) [inline src/utils/style-formatter.js]: Needs a new name since it is not returning a css property anymore. `getColor`?

**Comment by haxxmaxx** [comment]
I might have a refactor idea for the getColor. But thats just me being obsessed with optimizing code 🙃

**Comment by niekvanstaveren** [comment]
> I might have a refactor idea for the getColor. But thats just me being obsessed with optimizing code 🙃

Any optimisation is welcome of course! You want to change things in here or open a new branch for that?

**Comment by haxxmaxx** [comment]
Naw just get it in, I'll try it on my branch

**Pull request merged**